compair/models/assignment.py

```diff
@@ -166,6 +169,28 @@ def after_comparing(self):
         else:
             return now >= self.compare_end.replace(tzinfo=pytz.utc)
 
+    @hybrid_property
+    def self_eval_period(self):
+        now = dateutil.parser.parse(datetime.datetime.utcnow().replace(tzinfo=pytz.utc).isoformat())
+        if not self.enable_self_evaluation:
+            return False
+        elif self.self_eval_start:
+            return self.self_eval_start.replace(tzinfo=pytz.utc) <= now < self.self_eval_end.replace(tzinfo=pytz.utc)
+        else:
+            if self.compare_start:
+                return now >= self.compare_start.replace(tzinfo=pytz.utc)
+            else:
+                return now >= self.answer_end.replace(tzinfo=pytz.utc)
+
+    @hybrid_property
+    def self_eval_grace(self):
+        now = dateutil.parser.parse(datetime.datetime.utcnow().replace(tzinfo=pytz.utc).isoformat())
```

Comment:
`self_eval_grace` should use the same logic as `self_eval_period` but with added 60 seconds (still using compare_grace)